AzureIotPnpDps.X/main.c

```diff
@@ -7,582 +7,1327 @@
+				RETURN_ERR_IF_FAILED(az_json_token_get_string(&jr.token, reboot_delay, sizeof(reboot_delay), NULL));
+				break;
+			}
+		}
+		else if (jr.token.kind == AZ_JSON_TOKEN_STRING)
+		{
+			RETURN_ERR_IF_FAILED(az_json_token_get_string(&jr.token, reboot_delay, sizeof(reboot_delay), NULL));
+			break;
+		}
+		RETURN_ERR_IF_FAILED(az_json_reader_next_token(&jr));
+	}
+
+	if (reboot_delay[0] != 'P' || reboot_delay[1] != 'T' || reboot_delay[strlen(reboot_delay) - 1] != 'S')
+	{
+		debug_printError("  MAIN: Reboot Delay wrong format");
+		RETURN_ERR_IF_FAILED(build_error_response_payload(response, span_error_no_payload_reboot, out_response));
+		return AZ_ERROR_ARG;
+	}
+
+	reboot_delay_seconds = atoi((const char *)&reboot_delay[2]);
+
+	start_json_object(&jw, response);
+	append_jason_property_string(&jw, span_command_reboot_status, span_command_reboot_success);
+	append_jason_property_int32(&jw, span_command_reboot_delay, reboot_delay_seconds);
+	end_json_object(&jw);
+
+	*out_response = az_json_writer_get_bytes_used_in_destination(&jw);
+
+	return AZ_OK;
 }
 
-static int send_telemetry_message(void)
+/**********************************************
+* Handle commands
+**********************************************/
+static void handle_command_message(
+				az_span payload,
+				az_iot_pnp_client_command_request* command_request)
 {
-  int rc = az_iot_pnp_client_telemetry_get_publish_topic(
-              &pnp_client, AZ_SPAN_EMPTY, NULL, telemetry_topic, sizeof(telemetry_topic), NULL);
-  RETURN_ERR_WITH_MESSAGE_IF_FAILED(rc, "Could not get the telemetry topic");
+	az_span command_response_span = AZ_SPAN_FROM_BUFFER(commands_response_payload);
+	uint16_t return_code;
+	az_result azRet;
+	int ret;
+
+	if (az_span_is_content_equal(span_command_getmaxmin, command_request->command_name))
+	{
+		azRet = process_getMaxMinReport(payload, command_response_span, &command_response_span);
+		if (azRet != AZ_OK)
+		{
+			return_code = 404;
+		}
+		else
+		{
+			return_code = 200;
+		}
+
+		// Send command response with report as JSON payload
+		if ((ret = send_command_response(command_request, return_code, command_response_span)) != 0)
+		{
+			debug_printError("  MAIN: Unable to send command response, az_result %x status %d", azRet, ret);
+		}
+	}
+	else if (az_span_is_content_equal(span_command_reboot, command_request->command_name))
+	{
+		azRet = process_reboot(payload, command_response_span, &command_response_span);
+
+		if (azRet != AZ_OK)
+		{
+			debug_printError("  MAIN: process_reboot() failed.  AZ Result %x %d", azRet, az_span_size(command_response_span));
+			if(az_span_size(command_response_span) == 0)
+			{
+				// if response is empty, payload was not in the right format.
+				if (az_result_failed(azRet = build_error_response_payload(command_response_span, span_error_no_payload_reboot, &command_response_span)))
+				{
+					debug_printError("  MAIN: Failed to build error response. AZ Result %d", azRet);
```

Comment:
we've used `0x%08x` [in the past](https://github.com/Azure/azure-sdk-for-c/blob/3d9f1a3d530ef5fa12cf481ce7e6a4b798e286d5/sdk/samples/iot/iot_sample_common.c#L68) which works well to correlate to an az_result value.
```suggestion
					debug_printError("  MAIN: Failed to build error response. AZ Result 0x%08x", azRet);
```

Reply:
changed all instances with 0x%08x

[commit](https://github.com/daisukeiot/Microchip-PIC-IoT-Wx-1/commit/e8683cbe4bce84d1a46556a24856d82750f067fa)